lib/sync/default-dataHandlers.js

```diff
@@ -0,0 +1,79 @@
+var syncUtil = require('../sync-util');
+var util = require('util');
+
+var COLL_POSTFIX = '_collision';
+var mongo;
+
+module.exports = function (db) {
+
+  if (!db) {
+    throw new Error('MongoDB instance must be passed to module.');
+  }
+  mongo = db
+
+  return {
+    doList: function (dataset_id, params, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doList : ' + dataset_id +
+                     ' :: query_params=' + util.inspect(params) +
+                     ' :: meta_data=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).find(params).toArray(cb);
+    },
+
+    doCreate: function (dataset_id, data, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doCreate : ' + dataset_id + ' :: ' + util.inspect(data) +
+                     ' :: meta=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).insertOne(data, cb);
+    },
+
+    doRead: function (dataset_id, uid, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doRead : ' + dataset_id +
+                      ' :: ' + uid +
+                      ' :: meta=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).findOne({"_id" : uid}, cb);
+    },
+
+    doUpdate: function (dataset_id, uid, data, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doUpdate : ' + dataset_id +
+                     ' :: ' + uid + ' :: ' + util.inspect(data) +
+                     ' :: meta=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).update({"_id" : uid}, data, cb);
```

Comment:
looks like `update` is being deprecated. Might be better using `updateOne` instead.